src/java/picard/analysis/TheoreticalSensitivity.java

```diff
@@ -0,0 +1,191 @@
+        log.info("Calculating quality sums from quality sampler");
+        final List<ArrayList<Integer>> qualitySums = qualitySampler.sampleCumulativeSums(N, sampleSize);
+
+        //if a quality sum of m qualities exceeds the quality sum threshold for n total reads, a SNP is called
+        final ArrayList<Double> qualitySumThresholds = new ArrayList<Double>(N);
+        for (int n = 0; n < N; n++) qualitySumThresholds.add(10*(n*Math.log10(2) + logOddsThreshold));
+
+
+        //probabilityToExceedThreshold[m][n] is the probability that the sum of m quality score
+        //exceeds the nth quality sum threshold
+        log.info("Calculating theoretical het sensitivity");
+        final List<ArrayList<Double>> probabilityToExceedThreshold = proportionsAboveThresholds(qualitySums, qualitySumThresholds);
+        final List<ArrayList<Double>> altDepthDistribution = hetAltDepthDistribution(N);
+        double result = 0.0;
+        for (int n = 0; n < N; n++) {
+            for (int m = 0; m <= n; m++) {
+                result += depthDistribution[n] * altDepthDistribution.get(n).get(m)* probabilityToExceedThreshold.get(m).get(n);
+            }
+        }
+        return result;
+    }
+
+    //given L lists of lists and N thresholds, count the proportion of each list above each threshold
+    public static List<ArrayList<Double>> proportionsAboveThresholds(final List<ArrayList<Integer>> lists, final List<Double> thresholds) {
+        final ArrayList<ArrayList<Double>> result = new ArrayList<ArrayList<Double>>();
+
+        for (final ArrayList<Integer> list : lists) {
+            final ArrayList<Double> newRow = new ArrayList<Double>(Collections.nCopies(thresholds.size(),0.0));
+            Collections.sort(list);
+            int n = 0;
+            int j = 0;  //index within the ordered sample
+            while (n < thresholds.size() && j < list.size()) {
+                if (thresholds.get(n) > list.get(j)) j++;
+                else newRow.set(n++, (double) (list.size() - j)/list.size());
+            }
+            result.add(newRow);
+        }
+        return result;
+
+    }
+
+
+    //Utility function for making table of binomial distribution probabilities nCm * (0.5)^n
+    //for n = 0, 1 . . . N - 1 and m = 0, 1. . . n
+    public static List<ArrayList<Double>> hetAltDepthDistribution(final int N) {
+        final List<ArrayList<Double>> table = new ArrayList<ArrayList<Double>>();
+        for (int n = 0; n < N; n++) {
+            final ArrayList<Double> nthRow = new ArrayList<Double>();
+
+            //add the 0th element, then elements 1 through n - 1, then the nth.
+            //Note that nCm = (n-1)C(m-1) * (n/m)
+            nthRow.add(Math.pow(0.5, n));
+            for (int m = 1; m < n; m++) nthRow.add((n*0.5/m)*table.get(n - 1).get(m - 1));
+            if (n > 0) nthRow.add(nthRow.get(0));
+
+            table.add(nthRow);
+        }
+
+        return table;
+
```

Comment:
I was apparently the Santa Claus of extraneous whitespace when I wrote this code. Sorry!